Tests/KodableTests.swift

```diff
@@ -637,6 +673,43 @@ final class KodableTests: XCTestCase {
         assert(try Dates.decodeJSON(from: KodableTests.json), throws: KodableError.failedToParseDate(source: "123456789987654321"))
     }
 
+    // MARK: - Equatable
+
+    func testCodableDateConformsToEquatable() throws {
+        struct RFCDate: Kodable, Equatable {
+            @CodableDate(.rfc2822, "rfc2822") var date: Date
+
+            static func fromJSON() throws -> RFCDate {
+                try RFCDate.decodeJSON(from: KodableTests.json)
+            }
+
+            static func now() -> RFCDate {
+                let now = RFCDate()
+                now.date = Date()
+                return now
+            }
+        }
+
+        let a = try RFCDate.fromJSON()
+        let b = RFCDate.now()
+        let c = a
+
+        XCTAssertNotEqual(a.date, b.date)
+        XCTAssertNotEqual(a, b)
+        XCTAssertEqual(a.date, c.date)
+        XCTAssertEqual(a, c)
+    }
+
+    // MARK: - OptionalProtocol Tests
+
+    func testOptionalProtocolIsNil() {
```

Comment:
Can you check if this test fails with nested optionals (e.g. `String??` aka `Optional<Optional<String>>`)? 

See https://gist.github.com/rogerluan/ee04febd80371f88f9435e98032b3042 for some 💡 😁

Reply:
updated